diff --git a/distributedaudio.gni b/distributedaudio.gni
index d2f03bd..a28ea6d 100644
--- a/distributedaudio.gni
+++ b/distributedaudio.gni
@@ -23,6 +23,8 @@ audio_framework_path =
     "//foundation/multimedia/audio_framework/frameworks/native"
 driver_audio_path = "//drivers/peripheral/audio/interfaces"
 
+distributed_audio_hdf_path = "//device/soc/hisilicon/common/hal/media/audio/source/distributed_audio"
+
 common_path = "${distributedaudio_path}/common"
 services_path = "${distributedaudio_path}/services"
 audio_client_path = "${services_path}/audioclient"
@@ -30,12 +32,18 @@ audio_control_path = "${services_path}/audiocontrol"
 audio_hdi_proxy_path = "${services_path}/audiohdiproxy"
 audio_processor_path = "${services_path}/audioprocessor"
 audio_transport_path = "${services_path}/audiotransport"
+softbusadapter_path = "${services_path}/softbusadapter"
 interfaces_path = "${distributedaudio_path}/interfaces"
 hdf_interfaces_path = "//drivers/interface/distributed_audio"
 hdf_service_path = "//drivers/peripheral/distributed_audio"
 hdf_ser_aud_path = "${hdf_service_path}/hdi_service/audio/v1_0"
 hdf_ser_aud_ext_path = "${hdf_service_path}/hdi_service/audio_ext/v1_0"
 
+hdf_service_path_lite = "${distributed_audio_hdf_path}/hdf_service/distributed_audio"
+hdf_ser_aud_path_lite = "${hdf_service_path_lite}/hdi_service/audio/v1_0"
+hdf_ser_aud_ext_path_lite = "${hdf_service_path_lite}/hdi_service/audio_ext/v1_0"
+
+
 fwk_utils_path = "${distributedhardwarefwk_path}/utils"
 fwk_common_path = "${distributedhardwarefwk_path}/common"
 fwk_services_path = "${distributedhardwarefwk_path}/services"
