diff --git a/data/SetPreparation.py b/data/SetPreparation.py
index 1922fbd..1b61ff5 100644
--- a/data/SetPreparation.py
+++ b/data/SetPreparation.py
@@ -62,7 +62,7 @@ class ImageDate():
                 assert len(lines) == 1
                 mirror_idx = lines[0].strip().split(',')
                 mirror_idx = list(map(int, mirror_idx))
-        xy = np.min(self.landmark, axis=0).astype(np.int32) 
+        xy = np.min(self.landmark, axis=0).astype(np.int32)
         zz = np.max(self.landmark, axis=0).astype(np.int32)
         wh = zz - xy + 1
 
@@ -110,7 +110,7 @@ class ImageDate():
 
                 imgT = cv2.warpAffine(img, M, (int(img.shape[1]*1.1), int(img.shape[0]*1.1)))
 
-                
+
                 wh = np.ptp(landmark, axis=0).astype(np.int32) + 1
                 size = np.random.randint(int(np.min(wh)), np.ceil(np.max(wh) * 1.25))
                 xy = np.asarray((cx - size // 2, cy - size//2), dtype=np.int32)
@@ -152,9 +152,9 @@ class ImageDate():
         TRACKED_POINTS = [33, 38, 50, 46, 60, 64, 68, 72, 55, 59, 76, 82, 85, 16]
         for i, (img, lanmark) in enumerate(zip(self.imgs, self.landmarks)):
             assert lanmark.shape == (98, 2)
-            save_path = os.path.join(path, prefix+'_'+str(i)+'.png')
+            save_path = os.path.join(path, prefix+'_'+str(i)+'.jpg')  # 改：保存为JPG格式（把.png改成.jpg）
             assert not os.path.exists(save_path), save_path
-            cv2.imwrite(save_path, img)
+            cv2.imwrite(save_path, img, [cv2.IMWRITE_JPEG_QUALITY, 95])  # 95是JPG质量（0-100）
 
             euler_angles_landmark = []
             for index in TRACKED_POINTS:
@@ -197,13 +197,12 @@ def get_dataset_list(imgDir, outDir, landmarkDir, is_train):
 
 if __name__ == '__main__':
     root_dir = os.path.dirname(os.path.realpath(__file__))
-    imageDirs = 'WFLW/WFLW_images'
-    Mirror_file = 'WFLW/WFLW_annotations/Mirror98.txt'
+    imageDirs = '../../../../../../datasets/WFLW/WFLW_images'
+    Mirror_file = '../../../../../../datasets/WFLW/WFLW_annotations/Mirror98.txt'
 
-    landmarkDirs = ['WFLW/WFLW_annotations/list_98pt_rect_attr_train_test/list_98pt_rect_attr_test.txt',
-                    'WFLW/WFLW_annotations/list_98pt_rect_attr_train_test/list_98pt_rect_attr_train.txt']
+    landmarkDirs = ['../../../../../../datasets/WFLW/WFLW_annotations/list_98pt_rect_attr_train_test/list_98pt_rect_attr_test.txt']
 
-    outDirs = ['test_data', 'train_data']
+    outDirs = ['../../data/wflw_test_cropped']
     for landmarkDir, outDir in zip(landmarkDirs, outDirs):
         outDir = os.path.join(root_dir, outDir)
         print(outDir)
diff --git a/dataset/datasets.py b/dataset/datasets.py
index e138996..338ef86 100644
--- a/dataset/datasets.py
+++ b/dataset/datasets.py
@@ -165,13 +165,14 @@ class WLFWDatasets(data.Dataset):
 
     def __getitem__(self, index):
         self.line = self.lines[index].strip().split()
+        self.path = self.line[0]
         self.img = cv2.imread(self.line[0])
         self.landmark = np.asarray(self.line[1:197], dtype=np.float32)
         self.attribute = np.asarray(self.line[197:203], dtype=np.int32)
         self.euler_angle = np.asarray(self.line[203:206], dtype=np.float32)
         if self.transforms:
             self.img = self.transforms(self.img)
-        return (self.img, self.landmark, self.attribute, self.euler_angle)
+        return (self.path, self.img, self.landmark, self.attribute, self.euler_angle)
 
     def __len__(self):
         return len(self.lines)
@@ -185,7 +186,7 @@ if __name__ == '__main__':
                             shuffle=True,
                             num_workers=0,
                             drop_last=False)
-    for img, landmark, attribute, euler_angle in dataloader:
+    for _, img, landmark, attribute, euler_angle in dataloader:
         print("img shape", img.shape)
         print("landmark size", landmark.size())
         print("attrbute size", attribute)
diff --git a/pytorch2onnx.py b/pytorch2onnx.py
index 915d741..4e35a76 100644
--- a/pytorch2onnx.py
+++ b/pytorch2onnx.py
@@ -32,6 +32,7 @@ torch.onnx.export(pfld_backbone,
                   dummy_input,
                   args.onnx_model,
                   verbose=True,
+                  opset_version=11,
                   input_names=input_names,
                   output_names=output_names)
 
@@ -41,7 +42,7 @@ model = onnx.load(args.onnx_model)
 onnx.checker.check_model(model)
 
 print("====> Simplifying...")
-model_opt = onnxsim.simplify(args.onnx_model)
+model_opt, check_ok = onnxsim.simplify(args.onnx_model)
 # print("model_opt", model_opt)
 onnx.save(model_opt, args.onnx_model_sim)
 print("onnx model simplify Ok!")
\ No newline at end of file
diff --git a/test.py b/test.py
index a8e30d8..32891e6 100644
--- a/test.py
+++ b/test.py
@@ -73,7 +73,7 @@ def validate(wlfw_val_dataloader, pfld_backbone):
     nme_list = []
     cost_time = []
     with torch.no_grad():
-        for img, landmark_gt, _, _ in wlfw_val_dataloader:
+        for path, img, landmark_gt, _, _ in wlfw_val_dataloader:
             img = img.to(device)
             landmark_gt = landmark_gt.to(device)
             pfld_backbone = pfld_backbone.to(device)
diff --git a/train.py b/train.py
index 3183df4..cbb82a8 100644
--- a/train.py
+++ b/train.py
@@ -75,7 +75,7 @@ def validate(wlfw_val_dataloader, pfld_backbone, auxiliarynet, criterion):
     auxiliarynet.eval()
     losses = []
     with torch.no_grad():
-        for img, landmark_gt, attribute_gt, euler_angle_gt in wlfw_val_dataloader:
+        for _, img, landmark_gt, attribute_gt, euler_angle_gt in wlfw_val_dataloader:
             img = img.to(device)
             attribute_gt = attribute_gt.to(device)
             landmark_gt = landmark_gt.to(device)
