diff --git a/interfaces/innerkits/appverify_lite/include/app_common.h b/interfaces/innerkits/appverify_lite/include/app_common.h
index 1d7e144..58f5dad 100644
--- a/interfaces/innerkits/appverify_lite/include/app_common.h
+++ b/interfaces/innerkits/appverify_lite/include/app_common.h
@@ -17,6 +17,7 @@
 #define SECURITY_APP_COMMON_H
 
 #include <stdint.h>
+#include <malloc.h>
 
 #include "app_verify_base.h"
 #include "app_verify_pub.h"
diff --git a/interfaces/innerkits/appverify_lite/src/app_verify.c b/interfaces/innerkits/appverify_lite/src/app_verify.c
index f84a6f6..39501eb 100644
--- a/interfaces/innerkits/appverify_lite/src/app_verify.c
+++ b/interfaces/innerkits/appverify_lite/src/app_verify.c
@@ -547,7 +547,7 @@ EXIT:
     APPV_FREE(profileData);
     return V_ERR;
 }
-static unsigned char *GetRsaPk(const mbedtls_pk_context *pk, int32_t *len)
+static unsigned char *GetRsaPk(const mbedtls_pk_context *pk, size_t *len)
 {
     unsigned char *buf = APPV_MALLOC(MAX_PK_BUF);
     if (buf == NULL) {
@@ -609,7 +609,7 @@ static unsigned char *GetEcPk(const mbedtls_pk_context *pk, int32_t *len)
         return NULL;
     }
     ret = mbedtls_ecp_point_write_binary(&ecCtx->MBEDTLS_PRIVATE(grp), &ecCtx->MBEDTLS_PRIVATE(Q),
-        MBEDTLS_ECP_PF_UNCOMPRESSED, (size_t *)len, buf, MBEDTLS_ECP_MAX_PT_LEN);
+        MBEDTLS_ECP_PF_UNCOMPRESSED, len, buf, MBEDTLS_ECP_MAX_PT_LEN);
     if (ret != V_OK) {
         LOG_ERROR("get ecc pk key error");
         (void)memset_s(buf, MBEDTLS_ECP_MAX_PT_LEN, 0, MBEDTLS_ECP_MAX_PT_LEN);
@@ -640,7 +640,7 @@ static unsigned char *GetEcPk(const mbedtls_pk_context *pk, int32_t *len)
     return pkBuf;
 }
 
-static unsigned char *GetPkBuf(const mbedtls_pk_context *pk, int32_t *len)
+static unsigned char *GetPkBuf(const mbedtls_pk_context *pk, size_t *len)
 {
     unsigned char *bufA = NULL;
     if (mbedtls_pk_get_type(pk) == MBEDTLS_PK_RSA || mbedtls_pk_get_type(pk) == MBEDTLS_PK_RSASSA_PSS) {
@@ -663,7 +663,7 @@ static int32_t ParseCertGetPk(const char *certEncoded, AppSignPk *pk)
         APPV_FREE(cert);
         return V_ERR;
     }
-    int32_t len = 0;
+    size_t len = 0;
     unsigned char *pkBuf = GetPkBuf(&cert->pk, &len);
     if (pkBuf == NULL) {
         LOG_ERROR("get pk error");
@@ -672,7 +672,7 @@ static int32_t ParseCertGetPk(const char *certEncoded, AppSignPk *pk)
         return V_ERR;
     }
     pk->pk = (char *)pkBuf;
-    pk->len = len;
+    pk->len = (int32_t)len;
     mbedtls_x509_crt_free(cert);
     APPV_FREE(cert);
     return V_OK;
@@ -814,7 +814,7 @@ static int32_t CmpCert(const mbedtls_x509_crt *certA, const CertInfo *binSignCer
         LOG_ERROR("pk type diff");
         return V_ERR;
     }
-    int32_t lenA = 0;
+    size_t lenA = 0;
     unsigned char *bufA = GetPkBuf(&certA->pk, &lenA);
     P_NULL_RETURN_RET_WTTH_LOG(bufA, V_ERR);
 
@@ -992,13 +992,15 @@ static int32_t GetCertInfo(const mbedtls_x509_crt *ctr, CertInfo **binSignCert)
         ret = V_ERR_MEMCPY;
         goto EXIT;
     }
+    size_t pkLen = 0;
     certInfo->pkType = mbedtls_pk_get_type(&ctr->pk);
-    certInfo->pkBuf = (char *)GetPkBuf(&ctr->pk, &certInfo->pkLen);
+    certInfo->pkBuf = (char *)GetPkBuf(&ctr->pk, &pkLen);
     if (certInfo->pkBuf == NULL) {
         LOG_ERROR("get pk error");
         ret = V_ERR;
         goto EXIT;
     }
+    certInfo->pkLen = (int32_t)pkLen;
     *binSignCert = certInfo;
     return V_OK;
 EXIT:
