diff --git a/services/bundlemgr_lite/src/hap_sign_verify.cpp b/services/bundlemgr_lite/src/hap_sign_verify.cpp
index b639829..c346279 100644
--- a/services/bundlemgr_lite/src/hap_sign_verify.cpp
+++ b/services/bundlemgr_lite/src/hap_sign_verify.cpp
@@ -12,20 +12,56 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
+#include <string>
+#include <unistd.h>
+#include <iostream>
 #include "hap_sign_verify.h"
 
 #include "appexecfwk_errors.h"
 #include "bundle_manager_service.h"
 #include "bundle_log.h"
 
+#define SKIP_APPVERIFY 0
+
 namespace OHOS {
+
+#if SKIP_APPVERIFY
+uint8_t HapSignVerify::VerifySignature(const std::string &hapFilepath, SignatureInfo &signatureInfo)
+{
+    bool mode = ManagerService::GetInstance().IsDebugMode();
+    HILOG_INFO(HILOG_MODULE_APP, "current mode is %d!", mode);
+    VerifyResult verifyResult;
+    // verify signature
+    HILOG_INFO(HILOG_MODULE_APP, "sf0827 hapFilepath %s!", hapFilepath.c_str());
+    if (hapFilepath == std::string("/system/internal/launcher.hap")) {
+        signatureInfo.appId = std::string("com.huawei.launcher_1errff456");
+        signatureInfo.provisionBundleName = std::string("com.huawei.launcher");
+    } else if (hapFilepath == std::string("/system/internal/setting.hap")) {
+        signatureInfo.appId = std::string("com.huawei.setting_3egrff486");
+        signatureInfo.provisionBundleName = std::string("com.huawei.setting");
+    } else if (hapFilepath == std::string("/system/internal/gallery.hap")) {
+        signatureInfo.appId = std::string("com.huawei.gallery_6erddf436");
+        signatureInfo.provisionBundleName = std::string("com.huawei.gallery");
+    } else if (hapFilepath == std::string("/system/internal/screensaver.hap")) {
+        signatureInfo.appId = std::string("com.huawei.screensaver_9erdee435");
+        signatureInfo.provisionBundleName = std::string("com.huawei.screensaver");
+    } else if (hapFilepath == std::string("/system/internal/cameraApp.hap")) {
+        signatureInfo.appId = std::string("com.huawei.cameraApp_6erdet425");
+        signatureInfo.provisionBundleName = std::string("com.huawei.cameraApp");
+    } else if (hapFilepath == std::string("/system/internal/entry-default-signed.hap")) {
+        signatureInfo.appId = std::string("com.rkh.myapplication_9dddet215");
+        signatureInfo.provisionBundleName = std::string("com.rkh.myapplication");
+    }
+    return ERR_OK;
+}
+#else
 uint8_t HapSignVerify::VerifySignature(const std::string &hapFilepath, SignatureInfo &signatureInfo)
 {
     bool mode = ManagerService::GetInstance().IsDebugMode();
     HILOG_INFO(HILOG_MODULE_APP, "current mode is %d!", mode);
     VerifyResult verifyResult;
     // verify signature
+     HILOG_INFO(HILOG_MODULE_APP, "hapFilepath %s!", hapFilepath.c_str());
     int32_t ret = APPVERI_AppVerify(hapFilepath.c_str(), &verifyResult);
     uint8_t errorCode = SwitchErrorCode(ret);
     if (errorCode != ERR_OK) {
@@ -41,6 +77,7 @@ uint8_t HapSignVerify::VerifySignature(const std::string &hapFilepath, Signature
     APPVERI_FreeVerifyRst(&verifyResult);
     return ERR_OK;
 }
+#endif
 
 uint8_t HapSignVerify::SwitchErrorCode(int32_t errorCode)
 {
