diff --git a/BUILD.gn b/BUILD.gn
index e658c77..d9e0349 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -47,7 +47,7 @@ ohos_executable("syscap_tool_bin") {
   if (is_mingw) {
     defines += [ "_POSIX_" ]
   }
-  if (defined(ohos_lite) && ohos_kernel_type == "liteos_m") {
+  if (defined(ohos_lite) && (ohos_kernel_type == "liteos_m" || ohos_kernel_type == "linux")) {
     cflags = [ "-DPATH_MAX=1024" ]
     cflags += [ "-Wno-int-conversion" ]
   }
